xgi/generators/random.py

```diff
@@ -76,13 +82,74 @@ def random_hypergraph(N, ps, order=None, seed=None):
         else:
             d = i + 1  # order, ps[0] is prob of edges (d=1)
 
-        potential_edges = combinations(nodes, d + 1)
-        n_comb = comb(N, d + 1, exact=True)
-        mask = np.random.random(size=n_comb) <= p  # True if edge to keep
+        size = d + 1
+        if p > 0:
+            h_uniform = uniform_erdos_renyi_hypergraph(n, size, p, seed=seed)
+            H << h_uniform
+
+    return H
+
+
+def random_hypergraph(N, ps, seed=None):
+    """Generates a random hypergraph
+
+    Generate N nodes, and connect any d+1 nodes
+    by a hyperedge with probability ps[d-1].
+
+    Parameters
+    ----------
+    N : int
+        Number of nodes
+    ps : list of float
+        List of probabilities (between 0 and 1) to create a
+        hyperedge at each order d between any d+1 nodes. For example,
+        ps[0] is the wiring probability of any edge (2 nodes), ps[1]
+        of any triangles (3 nodes).
+    seed : integer, random_state, or None (default)
+            Indicator of random number generation state.
+
+    Returns
+    -------
+    Hypergraph object
+        The generated hypergraph
+
+    References
+    ----------
+    Described as 'random hypergraph' by M. Dewar et al. in https://arxiv.org/abs/1703.07686
+
+    Warns
+    -----
+    warnings.warn
+        Because `fast_random_hypergraph` is a much faster method for generating random hypergraphs.
+
+    See Also
+    --------
+    fast_random_hypergraph
+
+    Example
+    -------
+    >>> import xgi
+    >>> H = xgi.random_hypergraph(50, [0.1, 0.01])
+
+    """
+    warn("This method is much slower than fast_random_hypergraph")
+    if seed is not None:
+        random.seed(seed)
+
+    if (np.any(np.array(ps) < 0)) or (np.any(np.array(ps) > 1)):
+        raise ValueError("All elements of ps must be between 0 and 1 included.")
+
+    nodes = range(N)  # list(G.nodes())
+    hyperedges = []  # hyperedges = list(G.edges())
+
+    for i, p in enumerate(ps):
+        d = i + 1  # order, ps[0] is prob of edges (d=1)
 
-        edges_to_add = [e for e, val in zip(potential_edges, mask) if val]
+        for hyperedge in combinations(nodes, d + 1):
+            if random.random() <= p:
+                hyperedges.append(hyperedge)
 
-        hyperedges += edges_to_add
+    hyperedges += [[i] for i in nodes]  # add singleton edges
```

Comment:
I think in generative models we don't add singletons as a general rule by default, so I'd remove?

Reply:
This was the behavior of the original function, but yeah, agreed, let's axe this.